src/rhash/hash.c

```diff
@@ -1143,6 +1143,76 @@ static int rc_hash_nintendo_ds(char hash[33], const char* path)
   return rc_hash_finalize(&md5, hash);
 }
 
+static uint32_t rc_read_bigendian_u32_value(void* file_handle)
+{
+  char le_buffer[4];
+  char be_buffer[4];
+  rc_file_read(file_handle, le_buffer, 4);
+  for (int ix = 0; ix < 4; ix++)
+  {
+    be_buffer[3 - ix] = le_buffer[ix];
+  }
+  return *((uint32_t*)be_buffer);
+}
+
+static int rc_hash_gamecube(char hash[33], const char* path)
+{
+  md5_state_t md5;
+  void* file_handle;
+  const int BASE_HEADER_SIZE = 0x2440;
```

Comment:
You should have some way to identify that the file is indeed a GameCube disc. If you pass a PSX disc to this function, it'll still try to return a hash, and since the addresses you're using for sizes could have anything in them, it could cause huge memory allocations.

Is there some constant identifier in the app_loader_header or BootDOL stuff?